emily/lambda/src/resources/deposits.rs

```diff
@@ -0,0 +1,487 @@
+    fn from(deposit_table_entry: &DepositTableEntry) -> Self {
+        let mut item: HashMap<String, AttributeValue> = HashMap::new();
+        item.insert("BitcoinTxid".to_string(), AttributeValue::S(deposit_table_entry.bitcoin_txid.clone()));
+        item.insert("BitcoinTxOutputIndex".to_string(), AttributeValue::N(deposit_table_entry.bitcoin_tx_output_index.to_string()));
+        item.insert("Version".to_string(), AttributeValue::N(0.to_string()));
+        item.insert("Recipient".to_string(), AttributeValue::S(deposit_table_entry.recipient.clone()));
+        item.insert("Amount".to_string(), AttributeValue::N(deposit_table_entry.amount.to_string()));
+        item.insert("MaxFee".to_string(), AttributeValue::N(deposit_table_entry.max_fee.to_string()));
+        item.insert("LockTime".to_string(), AttributeValue::S(deposit_table_entry.lock_time.to_string()));
+        item.insert("ReclaimScript".to_string(), AttributeValue::S(deposit_table_entry.reclaim_script.clone()));
+        item.insert("OpStatus".to_string(), AttributeValue::N(status_to_id(deposit_table_entry.op_status).to_string()));
+        item.insert("LastUpdateHeight".to_string(), AttributeValue::N(deposit_table_entry.last_update_height.to_string()));
+        item.insert("LastUpdateBlockHash".to_string(), AttributeValue::S(deposit_table_entry.last_update_block_hash.clone().to_string()));
+        if let Some(fulfillment_bitcoin_txid) = deposit_table_entry.fulfillment_bitcoin_txid.clone() {
+            item.insert("FulfillmentBitcoinTxid".to_string(), AttributeValue::S(fulfillment_bitcoin_txid.to_string()));
+        }
+        if let Some(fulfillment_bitcoin_tx_index) = deposit_table_entry.fulfillment_bitcoin_tx_index {
+            item.insert("FulfillmentBitcoinTxIndex".to_string(), AttributeValue::N(fulfillment_bitcoin_tx_index.to_string()));
+        }
+        if let Some(fulfillment_btc_fee) = deposit_table_entry.fulfillment_btc_fee {
+            item.insert("FulfillmentBtcFee".to_string(), AttributeValue::N(fulfillment_btc_fee.to_string()));
+        }
+        if let Some(stacks_txid) = deposit_table_entry.stacks_txid.clone() {
+            item.insert("StacksTxid".to_string(), AttributeValue::S(stacks_txid.to_string()));
+        }
+        item.insert("History".to_string(), AttributeValue::L(
+            deposit_table_entry.history.iter().map(|history_entry|
+            history_entry.into()
+        ).collect()));
+        item
+    }
+}
+
+/// Converts from Deposit Table Entry to Deposit Resource.
+impl From<DepositTableEntry> for DepositResource {
+    fn from(deposit_table_entry: DepositTableEntry) -> Self {
+        (&deposit_table_entry).into()
+    }
+}
+
+/// Converts from Deposit Table Entry to Deposit Resource.
+impl From<&DepositTableEntry> for DepositResource {
+    fn from(deposit_table_entry: &DepositTableEntry) -> Self {
+        let latest_history_entry = deposit_table_entry.history
+            .last() // There must always be at least one entry.
```

Comment:
How is this invariant enforced? I think we could use something like this crate https://crates.io/crates/nonempty - or define our own non empty Vec type, but I don't want to have a type conversion panic if we forget at some point in the code to maintain this invariant.